### ISSUE TITLE
descriptor with master fingerprint and descriptor key path triggers `InvalidKey` when signing

### ISSUE THREAD

**Issue by davemo88** [bitcoindevkit/bdk#307]
I think the `Signer::sign` implementation for `DescriptorXKey<ExtendedPrivKey>` gets confused by descriptors that include the master fingerprint / derivation path for the descriptor key.

For example:

`wpkh(tprvblahblahblah/0/*)` 
 
can sign fine but 

`wpkh([fingerprint/some/path/here]tprvblahblahblah/0/*)`

 triggers `InvalidKey`. If you include the master fingerprint and path, `get_hd_keypaths` returns the full path from the master, but then `sign` incorrectly uses the full path to derive from the descriptor key, instead of just the keychain / key index part.

I wrote [a test](https://github.com/bitcoindevkit/bdk/compare/master...davemo88:origin-bug) that shows the issue. It passes if you remove the fingerprint and path from `descriptor_key`.

**Comment by notmandatory** [comment]
I was also able to repo this issue with `bdk-cli`.

Unable to sign if using descriptor with origin:
```
➜ export ORIGIN_DESC="wpkh([d34db33f/84h/1h/0h]tprv8ZgxMBicQKsPd3EupYiPRhaMooHKUHJxNsTfYuScep13go8QFfHdtkG9nRkFGb7busX4isf6X9dURGCoKgitaApQ6MupRhZMcELAxTBRJgS/*)"
➜ bdk-cli wallet -d $ORIGIN_DESC get_new_address
{
  "address": "tb1q4ycgjh940hazv9jdwu9zm9avgrnk7twj96pg50"
}
➜  bdk-cli wallet -d $ORIGIN_DESC sync          
{}
➜  bdk-cli wallet -d $ORIGIN_DESC get_balance
{
  "satoshi": 10000
}
➜  bdk-cli wallet -d $ORIGIN_DESC create_tx -a --to tb1qm5tfegjevj27yvvna9elym9lnzcf0zraxgl8z2:0
{
  "details": {
    "fees": 110,
    "height": null,
    "received": 0,
    "sent": 10000,
    "timestamp": 1615601780,
    "transaction": null,
    "txid": "e46d5b86dd3cd54daf18a35196cd498882a31fb66b613f6a1da8a7474e605c5a"
  },
  "psbt": "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiBgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVBTTTbM/VAAAgAEAAIAAAACAAAAAAAAA"
}
➜  bdk-cli wallet -d $ORIGIN_DESC sign --psbt "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiBgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVBTTTbM/VAAAgAEAAIAAAACAAAAAAAAA"
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Signer(InvalidKey)', /home/steve/.cargo/registry/src/github.com-1ecc6299db9ec823/bdk-cli-0.2.0/src/bdk_cli.rs:177:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Able to sign and broadcast with same descriptor but no origin for xprv:
```
➜  export NO_ORIGIN_DESC="wpkh(tprv8ZgxMBicQKsPd3EupYiPRhaMooHKUHJxNsTfYuScep13go8QFfHdtkG9nRkFGb7busX4isf6X9dURGCoKgitaApQ6MupRhZMcELAxTBRJgS/*)" 
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC get_new_address
{
  "address": "tb1q4ycgjh940hazv9jdwu9zm9avgrnk7twj96pg50"
}
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC sync           
{}
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC get_balance       
{
  "satoshi": 10000
}
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC create_tx -a --to tb1qm5tfegjevj27yvvna9elym9lnzcf0zraxgl8z2:0
{
  "details": {
    "fees": 110,
    "height": null,
    "received": 0,
    "sent": 10000,
    "timestamp": 1615602415,
    "transaction": null,
    "txid": "e46d5b86dd3cd54daf18a35196cd498882a31fb66b613f6a1da8a7474e605c5a"
  },
  "psbt": "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiBgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVAi23/mQAAAAAAAA"
}
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC sign --psbt "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiBgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVAi23/mQAAAAAAAA"
{
  "is_finalized": true,
  "psbt": "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiAgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVEcwRAIgH2zdqB1tq98Kw1jDP60Dw665oKE6J3TNRjRM4u+zYkgCIEO3J5IQqu1l6Hlw7dZDgU+9xhKEb6E/R+h06FFKuUZSASIGAupEVb5HbreAOfwK4FuYAcuRJVowaL2vR6sYcMDmmmlUCLbf+ZAAAAAAAQcAAQhrAkcwRAIgH2zdqB1tq98Kw1jDP60Dw665oKE6J3TNRjRM4u+zYkgCIEO3J5IQqu1l6Hlw7dZDgU+9xhKEb6E/R+h06FFKuUZSASEC6kRVvkdut4A5/ArgW5gBy5ElWjBova9HqxhwwOaaaVQAAA=="
}
➜  bdk-cli wallet -w test2 -d $NO_ORIGIN_DESC broadcast --psbt "cHNidP8BAFIBAAAAAW1d2UrsIh8yWGV3CGnvS5n061V2DVra24sRwrlste3eAQAAAAD/////AaImAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBHxAnAAAAAAAAFgAUqTCJXLV9+iYWTXcKLZesQOdvLdIiAgLqRFW+R263gDn8CuBbmAHLkSVaMGi9r0erGHDA5pppVEcwRAIgH2zdqB1tq98Kw1jDP60Dw665oKE6J3TNRjRM4u+zYkgCIEO3J5IQqu1l6Hlw7dZDgU+9xhKEb6E/R+h06FFKuUZSASIGAupEVb5HbreAOfwK4FuYAcuRJVowaL2vR6sYcMDmmmlUCLbf+ZAAAAAAAQcAAQhrAkcwRAIgH2zdqB1tq98Kw1jDP60Dw665oKE6J3TNRjRM4u+zYkgCIEO3J5IQqu1l6Hlw7dZDgU+9xhKEb6E/R+h06FFKuUZSASEC6kRVvkdut4A5/ArgW5gBy5ElWjBova9HqxhwwOaaaVQAAA=="
{
  "txid": "e46d5b86dd3cd54daf18a35196cd498882a31fb66b613f6a1da8a7474e605c5a"
}
```

**Comment by afilini** [comment]
Fixed by #308
